clothing_app/clothing_app/middlewares.py

```diff
@@ -100,3 +105,99 @@ def process_exception(self, request, exception, spider):
 
     def spider_opened(self, spider):
         spider.logger.info("Spider opened: %s" % spider.name)
+
+
+class ClothingAppProxyMiddleware(object):
```

Comment:
Name this class as "RotateProxyMiddleware". It is only responsible for choosing a suitable proxy from a list. It maintains some weights and randomly selects that proxy.
It just sets `request['meta']['proxy'] = 'http:user:pass@ip:port'. Builtin proxy middleware will take care of settings auth headers etc.